contracts/0.4.24/lib/StakeLimitUtils.sol

```diff
@@ -94,17 +94,18 @@ library StakeLimitUnstructuredStorage {
 library StakeLimitUtils {
     /**
     * @notice Calculate stake limit for the current block.
+    * @dev using `_constGasMin` to make gas consumption independent of the current block number
     */
     function calculateCurrentStakeLimit(StakeLimitState.Data memory _data) internal view returns(uint256 limit) {
         uint256 stakeLimitIncPerBlock;
         if (_data.maxStakeLimitGrowthBlocks != 0) {
             stakeLimitIncPerBlock = _data.maxStakeLimit / _data.maxStakeLimitGrowthBlocks;
         }
 
-        limit = _data.prevStakeLimit + ((block.number - _data.prevStakeBlockNumber) * stakeLimitIncPerBlock);
-        if (limit > _data.maxStakeLimit) {
-            limit = _data.maxStakeLimit;
-        }
+        limit = _constGasMin(
+            _data.prevStakeLimit + ((block.number - _data.prevStakeBlockNumber) * stakeLimitIncPerBlock),
```

Comment:
Can probably use another variable with clear naming for this formula. It will improve readability a bit.

Reply:
Yeah, refined this part, thank you :+1: